### ISSUE TITLE
"Argument type is partially unknown" error  when using slicing assignment with an empty list

### ISSUE THREAD

**Issue by DetachHead** [microsoft/pyright#6994]
**Describe the bug**
when assigning an empty list using the `[:]` syntax, pyright fails to infer its type ([mypy is able to infer the type in this situation](https://mypy-play.net/?mypy=latest&python=3.12&flags=strict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Clocal-partial-types%2Cstrict-equality%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=99596cdb126c33a38ea84a421f4c887e))

**Code or Screenshots**

```python
a: list[int]
a = [] # no error
a[:] = [] # error: Argument type is partially unknown
```
[playground](https://pyright-play.net/?strict=true&code=IYLgBANglgzgLgbSgOzgXQLAChhgLxgJpgDEYyA9mAKYBOtFt2wCIxBRpN9j4AgrQDmAVwC21VGDgBPAA7UwsMLOC04UYBAjSww5AGtKAd2RA)

**Comment by erictraut** [comment]
Yes, this is intended behavior. 

The statement `a = []` is a simple assignment that doesn't involve any call.

The statement `a[:] = []` involves an implicit call to a magic method, and you are passing values with an `Unknown` type to that call, hence the `reportUnknownArgumentType` diagnostic.
